rocky/rocky/scheduler.py

```diff
@@ -310,6 +310,13 @@ def post_schedule(self, schedule: ScheduleRequest) -> ScheduleResponse:
         except (ValidationError, HTTPStatusError, ConnectError):
             raise SchedulerValidationError(extra_message="Report schedule failed: ")
 
+    def delete_schedule(self, schedule_id: str) -> None:
+        try:
+            response = self._client.delete(f"/schedules/{schedule_id}")
+            response.raise_for_status()
+        except (HTTPStatusError, ConnectError):
```

Comment:
A `ConnectError` seems quite specific in this case. Perhaps a broader `RequestError` would also suffice (or `TransportError` to handle connection errors and timeouts)

Reply:
I actually just copypasted `patch_schedule` and changed patch to delete without thinking about this. Most methods in the `SchedulerClient` don't catch all the httpx exceptions. I think in most places we should just catch `HTTPError`, but we should do this in a separate issue / PR.

Reply:
ideally we should not be dealing with the schedulerclient in both this and the other view. We should just be instantiating Schedule objects and calling on their crud methods. 
The Schedule object itself would then abstract away the interaction with the SchedulerClient and catch any errors that are HTTP specific.

Reply:
Good points. Maybe as an initial improvement to begin with, abstract away all the external dependency exceptions. For example, users that use a `SchedulerClient` shouldn't necessarily know or care about `httpx` and its exceptions, but they could expect a higher-level `SchedulerError` (that could optionally wrap an `httpx` exception for later reference) and its subclasses. And in the client's response hooks we could catch unhandled `httpx` exceptions and reraise a proper `SchedulerError`.

This could be an easy win for simplifying error management and improving maintainability and handle [this situation](https://github.com/minvws/nl-kat-coordination/issues/258#issuecomment-2084930396) since there are much less exceptions to handle, because we could just act on the generic `SchedulerError` exception and render the appropriate message.

Obviously this applies to other clients with the codebase and external dependencies.

Reply:
Agreed. The clients should (as we discussed before) just wrap the common errors in an error class for easy consumption by the user. the user shouldnt need to know if the client talks to a database, an api or a filesystem.